### PR TITLE
feat(engine): Apply output mappings for none end events

### LINKED ISSUE #10613: None end events calculation of output parameters
**Is your feature request related to a problem? Please describe.**

Process does not calculate output parameters in “none end event”. Is there a chance to calculate these output parameters and put them in the process values without using the “message end event”?

**Describe the solution you'd like**

The process instance applies the output mappings when a none end event is reached. Similar to message end events.

image

**Describe alternatives you've considered**
For now, using "none throw event" or "message end event" for calculate output parameters are alternative solutions.

**Additional context**
We are using Camunda Platform 8 Self Managed and Desktop Modeler

Related camunda forum link: https://forum.camunda.io/t/camunda-8-end-event-output-parameters/40136

The following class controls the behavior of the event events: [EndEventProcessor](https://github.com/camunda/zeebe/blob/main/engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/event/EndEventProcessor.java)

New contributors could watch the [video](https://www.twitch.tv/videos/1584245006) to learn more about how to extend the behavior of the end events.

### PR DESCRIPTION
**Pull request by skayliu**
## Description

Support none end event outputs.

## Related issues

closes #10613 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by skayliu** [comment]
Hi, @saig0, It's support none end event outputs, please reivew it, thank you.

**Comment by saig0** [comment]
@skayliu thank you for your contribution. :tada:  We'll have a look at your PR in the next few days. :eyes:

**Review by remcowesterhoud — CHANGES_REQUESTED** [review]
Thanks for your contribution @skayliu 🏅 

Overall the change looks fine. I have tested and it works as intended, so great job on that!

Unfortunately I can't approve it yet. The test is too limited as-is so we need to improve a bit here. I would recommend having a look at our other `XOutputMappingTest` classes to see which scenarios we've tested. Some examples can be found here:

- https://github.com/camunda/zeebe/blob/main/engine/src/test/java/io/camunda/zeebe/engine/processing/variable/mapping/MessageOutputMappingTest.java
- https://github.com/camunda/zeebe/blob/main/engine/src/test/java/io/camunda/zeebe/engine/processing/variable/mapping/JobOutputMappingTest.java

Note that these are parametrized tests so there's no need to write the same test case over and over again. Let me know if you need help with this!

**Comment by skayliu** [comment]
Hi, @remcowesterhoud, I've changed the test according to your opinion, Please review again, thank you.

File: engine/src/test/java/io/camunda/zeebe/engine/processing/variable/mapping/NoneEndEventOutputMappingTest.java
Comment (remcowesterhoud) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/variable/mapping/NoneEndEventOutputMappingTest.java:200]: ❌ You may have noticed executing these tests takes a long time. This is because we do not have a limit when searching for the `latestPayloadVariablePosition`. It's hard to define a proper limit for this, but we could make this simpler by getting the position of the end event `ACTIVATED` event instead.

```
    final long endEventActivatePosition =
        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
            .withProcessInstanceKey(processInstanceKey)
            .withElementId("end")
            .limit(1)
            .getFirst()
            .getPosition();
```

**Review by remcowesterhoud — CHANGES_REQUESTED** [review]
Thanks for your changes @skayliu 🚀 This is already a great improvement. There is 1 more thing that needs to change to speed up this test. After that I can merge it and set the `hacktoberfest-accepted` label 😄

**Review by remcowesterhoud — APPROVED** [review]
Great work! 🏆 

bors merge

**Comment by remcowesterhoud** [comment]
No there's no need. This is a new feature so we will just release as part of the the next minor version. Generally, we only backport bug fixes.

**Comment by saig0** [comment]
This pull request counts toward the Hacktoberfest challenge. If you contributed four pull requests then you can complete your Camunda Hacktoberfest challenge [here](https://camunda.com/hacktoberfest2022/).

**Pull request merged**
